pkg/txnutil/gc/gc_manager.go

```diff
@@ -109,20 +107,32 @@ func (m *gcManager) CheckStaleCheckpointTs(
 	ctx context.Context, changefeedID model.ChangeFeedID, checkpointTs model.Ts,
 ) error {
 	gcSafepointUpperBound := checkpointTs - 1
-	if m.isTiCDCBlockGC {
-		pdTime, err := m.pdClock.CurrentTime()
-		// TODO: should we return err here, or just log it?
-		if err != nil {
-			return errors.Trace(err)
-		}
-		if pdTime.Sub(oracle.GetTimeFromTS(gcSafepointUpperBound)) > time.Duration(m.gcTTL)*time.Second {
-			return cerror.ErrGCTTLExceeded.GenWithStackByArgs(checkpointTs, changefeedID)
-		}
-	} else {
-		// if `isTiCDCBlockGC` is false, it means there is another service gc point less than the min checkpoint ts.
-		if gcSafepointUpperBound < m.lastSafePointTs {
-			return cerror.ErrSnapshotLostByGC.GenWithStackByArgs(checkpointTs, m.lastSafePointTs)
-		}
+	// if there is another service gc point less than the min checkpoint ts.
+	if gcSafepointUpperBound < m.lastSafePointTs {
+		return cerror.ErrSnapshotLostByGC.
+			GenWithStackByArgs(
+				checkpointTs,
+				m.lastSafePointTs,
+			)
 	}
 	return nil
 }
+
+func (m *gcManager) IgnoreFailedFeed(
+	checkpointTs uint64,
+) bool {
+	pdTime, err := m.pdClock.CurrentTime()
+	if err != nil {
+		log.Warn("failed to get ts",
+			zap.String("GcManagerID", m.gcServiceID),
+			zap.Error(err),
+		)
+		return false
+	}
+	// ignore the change feed if its current checkpoint TS is earlier
+	// than the (currentPDTso - 24 hours).
+	gcSafepointUpperBound := checkpointTs - 1
+	return pdTime.Sub(
+		oracle.GetTimeFromTS(gcSafepointUpperBound),
+	) > 24*time.Hour
```

Comment:
prefer a const rather than a magic number

Reply:
define a const to replace the magic number